packages/components/src/components/Search/Search.tsx

```diff
@@ -1,35 +1,118 @@
+import { Animated, View } from "react-native";
+import { TextInput } from "react-native-gesture-handler";
+import { Button, Input, TextFieldProps, useStyles } from "../..";
+import { EDSStyleSheet } from "../../styling";
 
-export type SearchProps = Omit<TextFieldProps, "multiline">;
-
-export const Search = (props: SearchProps) => {
+export type SearchProps = Omit<TextFieldProps, "multiline"> & {
+    allowCancel?: boolean;
+    onCancelPress?: () => void;
+};
+export const Search = ({ allowCancel, onCancelPress, ...restProps }: SearchProps) => {
+    const [text, setText] = useState('');
+    const [isInputFocused, setIsInputFocused] = useState(false);
+    const [cancelButtonWidth, setCancelButtonWidth] = useState<number>(0);
+    const inputRef = useRef<TextInput | null>(null);
+    const animationValue = useRef(new Animated.Value(0)).current;
     const styles = useStyles(themedStyles);
+
+    const cancelButtonOpacity = animationValue.interpolate({
+        inputRange: [0, 1],
+        outputRange: [0, 1]
+    });
+
+    const cancelButtonTranslateX = animationValue.interpolate({
+        inputRange: [0, 1],
+        outputRange: [cancelButtonWidth, 0]
+    });
+
+    const inputSlide = animationValue.interpolate({
+        inputRange: [0, 1],
+        outputRange: [0, cancelButtonWidth]
+    });
+
+    useEffect(() => {
+        if (!allowCancel) return;
+        Animated.timing(animationValue, {
+            toValue: isInputFocused ? 1 : 0,
+            duration: 300,
+            useNativeDriver: false
+        }).start();
+    }, [isInputFocused]);
+
+
+    const handleCancel = () => {
+        setText('');
+        onCancelPress?.();
+        inputRef.current?.blur();
+    };
+
+    // FIXME: On web the search field loses focus when the clear text button is pressed. 
+    const handleClearText = () => {
+        setText('');
+        inputRef.current?.focus();
+    }
+
     return (
-        <Input
-            {...props}
-            leftAdornments={
-                <View style={styles.adornment}>
-                    <MaterialIcons name="search" size={18} color={styles.icon.color} />
-                </View>
-            }
-        ></Input>
+        <View style={styles.container}>
+            <Animated.View style={{ flex: 1, marginRight: inputSlide }}>
+                <Input
+                    {...restProps}
+                    ref={inputRef}
+                    value={text}
+                    onChange={setText}
+                    onFocus={() => setIsInputFocused(true)}
+                    onBlur={() => setIsInputFocused(false)}
+                    leftAdornments={
+                        <View style={styles.adornment}>
+                            <MaterialIcons name="search" size={18} color={styles.icon.color} />
+                        </View>
+                    }
+                    rightAdornments={
+                        text || allowCancel && text ? (
```

Comment:
is the `allowCancel && text` expression needed here? 

|`text`|`allowCancel`|__result__|
|---|---|---|
|0|0|0|
|0|1|0|
|1|0|1|
|1|1|1|

This is equivalent to just truth-checking `text`